cmd/cloud-controller-manager/controller-manager.go

```diff
@@ -17,32 +17,128 @@ limitations under the License.
+	"k8s.io/cloud-provider/options"
+	"k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/logs"
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
 	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
+	genericcontrollermanager "k8s.io/controller-manager/app"
+	"k8s.io/klog/v2"
+	nodeipamconfig "k8s.io/kubernetes/pkg/controller/nodeipam/config"
+)
+
+const (
+	// cloudProviderName shows an sample of using hard coded parameter
+	cloudProviderName = "SampleCloudProviderName"
+
+	// defaultNodeMaskCIDRIPv4 is default mask size for IPv4 node cidr
+	defaultNodeMaskCIDRIPv4 = 24
+	// defaultNodeMaskCIDRIPv6 is default mask size for IPv6 node cidr
+	defaultNodeMaskCIDRIPv6 = 64
 )
 
 func main() {
 	rand.Seed(time.Now().UnixNano())
 
-	command := app.NewCloudControllerManagerCommand()
+	// cloudProviderConfigFile shows an sample of parse config file from flag option
+	var flagset *pflag.FlagSet = pflag.NewFlagSet("flagSet", pflag.ContinueOnError)
+	var cloudProviderConfigFile *string = flagset.String("cloud-provider-configfile", "", "This is the sample input for cloud provider config file")
+	pflag.CommandLine.ParseErrorsWhitelist.UnknownFlags = true
+	_ = pflag.CommandLine.Parse(os.Args[1:])
+
+	// this is an example of allow-listing specific controller loops
+	controllerList := []string{"cloud-node", "cloud-node-lifecycle", "service", "route"}
+
+	s, err := options.NewCloudControllerManagerOptions()
+	if err != nil {
+		klog.Fatalf("unable to initialize command options: %v", err)
+	}
+	c, err := s.Config(controllerList, app.ControllersDisabledByDefault.List())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+
+	cloud, err := cloudprovider.InitCloudProvider(cloudProviderName, *cloudProviderConfigFile)
+	if err != nil {
+		klog.Fatalf("Cloud provider could not be initialized: %v", err)
+	}
+	if cloud == nil {
+		klog.Fatalf("cloud provider is nil")
+	}
+
+	if !cloud.HasClusterID() {
+		if c.ComponentConfig.KubeCloudShared.AllowUntaggedCloud {
+			klog.Warning("detected a cluster without a ClusterID.  A ClusterID will be required in the future.  Please tag your cluster to avoid any future issues")
+		} else {
+			klog.Fatalf("no ClusterID found.  A ClusterID is required for the cloud provider to function properly.  This check can be bypassed by setting the allow-untagged-cloud option")
+		}
+	}
+
+	// Initialize the cloud provider with a reference to the clientBuilder
+	cloud.Initialize(c.ClientBuilder, make(chan struct{}))
+	// Set the informer on the user cloud object
+	if informerUserCloud, ok := cloud.(cloudprovider.InformerUser); ok {
+		informerUserCloud.SetInformers(c.SharedInformers)
+	}
+
+	// remove the controller which current cloud provider does not need. e.g. cloud-node-lifecycle
+	controllerInitializers := app.DefaultControllerInitializers(c.Complete(), cloud)
+	delete(controllerInitializers, "cloud-node-lifecycle")
```

Comment:
this should probably be commented out before merge, just to prevent people from accidentally copy/pasting this and disabling the cloud-node-lifecycle controller accidentally

Reply:
Good point. Let me commented out and add more comment to make it clear